docs/source/routing/observability/debugging-client-requests.mdx

```diff
@@ -0,0 +1,82 @@
+
+## Using GraphOS Insights
+
+Without enabling additional telemetry, the easiest way to analyze GraphQL operation data is through [GraphOS Insights](/graphos/platform/insights). GraphOS Studio lets you analyze data from failed requests, such as GraphQL error messages ([if enabled](/graphos/routing/graphos-reporting#errors)) and the ID of the client making the request. You can also [segment your insights data](/graphos/platform/insights/client-segmentation) based on the client ID.
+
+<Tip>
+
+[Learn how to ensure client IDs are included in all requests.](/graphos/routing/observability/client-id-enforcement)
+
+</Tip>
+
+## Enabling additional telemetry
+
+Use the methods below if you need information outside of what's presented in GraphOS Studio to debug client requests.
+
+<Note>
+
+If you want to debug client requests in your own environment, Apollo recommends first doing so in a non-production environment or using logic to debug on a per-request basis.
+
+</Note>
+
+### Telemetry spans and traces
+
+Once you enable [router telemetry](/graphos/routing/observability/telemetry), traces have spans labeled `router`. These spans contain the HTTP information for a given client request via [OpenTelemetry standard attributes](/graphos/reference/router/telemetry/instrumentation/standard-attributes#router) such as `error.type` and `http.response.status_code`.
+
+These attributes include helpful information about the request and response. Spans, however, don't include an attribute for the HTTP body text. If you want to see the GraphQL operation, first check your [GraphOS Insights](/graphos/platform/insights) or use the other options listed below.
+
+### Request and response logging
+
+You can conditionally include the request body in your logs using the [`experimental-when-header`](/graphos/reference/router/telemetry/log-exporters/overview#requestresponse-logging) configuration option.
+
+```yaml title="router.yaml"
+telemetry:
+  exporters:
+    logging:
+      # If one of these headers matches we will log supergraph and subgraphs requests/responses
+      experimental_when_header:
+        - name: apollo-router-log-request
+          value: my_client
+          headers: true # default: false
+          body: true # default: false
+```
+
+### Debugging router logs
+
+By default, the router uses the `info` level for its logging. [Enabling other logging levels](/graphos/reference/router/telemetry/log-exporters/overview) can help debug specific scenarios. Using non-`info` level configurations is only recommended for local or non-production environments.
+
+### Rhai scripts and coprocessors
+
+Hooking into the `RouterService` layer with either [Rhai scripts](/graphos/routing/customization/rhai) or [coprocessors](/graphos/routing/customization/coprocessor) gives you access to the full HTTP request and response before processing occurs. You can use either Rhai scripts or coprocessors to add custom logic for what to log and when.
```

Comment:
@smyrick , the `before processing occurs` in this line confuses me. What processing are we talking about?

Reply:
The `RouterService` external hook is called before the Router does any GraphQL parsing, Query Planning, or other features like Auth checks, so this is the first place you can log out the exact request before anything could error out

Reply:
Gotcha, the "response" part of `gives you access to the full HTTP request and _response_ before processing occurs` was what was giving me pause. Will just remove that.

```suggestion
Hooking into the `RouterService` layer with either [Rhai scripts](/graphos/routing/customization/rhai) or [coprocessors](/graphos/routing/customization/coprocessor) gives you access to the full HTTP request before processing occurs. You can use either Rhai scripts or coprocessors to add custom logic for what to log and when.
```